docs/recipes/add-custom-chain.mdx

```diff
@@ -0,0 +1,163 @@
+      http: ["https://mainnet.base.org"],
+    },
+  },
+  blockExplorers: {
+    default: {
+      name: "Basescan",
+      url: "https://basescan.org",
+    },
+    etherscan: {
+      name: "Basescan",
+      url: "https://basescan.org",
+    },
+  },
+  contracts: {
+    multicall3: {
+      address: "0xca11bde05977b3631167028862be2a173976ca11",
+      blockCreated: 5022,
+    },
+  },
+});
+```
+
+In this file, we're defining the Base chain. We're using the `defineChain` function from viem to define the chain. You can add as many chains as you want to the `customChains.ts` file.
+
+### Step 2: Update `scaffold.config.ts`
+
+Next, update your `scaffold.config.ts` file to include the new chain:
+
+```typescript title="packages/nextjs/scaffold.config.ts"
+import { base } from "./utils/customChains";
+// ... other imports and type definitions
+
+const scaffoldConfig = {
+  targetNetworks: [base],
+  // ... other configuration options
+} as const satisfies ScaffoldConfig;
+
+export default scaffoldConfig;
+```
+
+If you'd like to add multiple chains, you can do so by adding them to the `targetNetworks` array. Below is a simple example of how to add multiple chains.
+
+```typescript title="packages/nextjs/scaffold.config.ts"
+import { base, baseSepolia } from "./utils/customChains";
+
+const scaffoldConfig = {
+  targetNetworks: [base, baseSepolia],
+  // ... other configuration options
+} as const satisfies ScaffoldConfig;
+```
+
+### Step 3: Update configuration
+
+To deploy to these new chains, you'll need to update your configuration. Add the following to your configuration file.
+
+```mdx-code-block
+<Tabs groupId="dev-tool">
+<TabItem value="hardhat" label="Hardhat" default>
+```
+
+```typescript title="packages/hardhat/hardhat.config.ts"
+networks: {
+    // ... other networks
+    base: {
+        url: "https://mainnet.base.org",
+        accounts: [deployerPrivateKey]
+    },
+}
+```
+
+If you'd like to add multiple chains, you can do so by adding them to the networks object.
+
+```typescript title="packages/hardhat/hardhat.config.ts"
+networks: {
+    // ... other networks
+    base: {
+        url: "https://mainnet.base.org",
+        accounts: [deployerPrivateKey]
+    },
+    baseSepolia: {
+        url: "https://sepolia.base.org",
+        accounts: [deployerPrivateKey]
+    },
+}
+```
+
+```mdx-code-block
+</TabItem>
+<TabItem value="foundry" label="Foundry">
+```
+
+```toml title="packages/foundry/foundry.toml"
+[rpc_endpoints]
+...other chains
+base = "https://mainnet.base.org"
+```
+
+If you'd like to add multiple chains, you can do so by adding them to the rpc_endpoints section.
+
+```toml title="packages/foundry/foundry.toml"
+[rpc_endpoints]
+...other chains
+base = "https://mainnet.base.org"
+base_sepolia = "https://sepolia.base.org"
+```
+
+</TabItem>
+</Tabs>
+
+### Step 4: Deploy to the new chain
+
+Now you can deploy your contracts to these new chains using the deploy command
+
+```bash
+yarn deploy --network base
+```
```

Comment:
I think we can remove this part? Since I feel it kind of does't fit into overall flow. Like people first deploy contract then want to update frontend to point to it. 

Maybe we can add a reference link to deploy-smart-contracts page? Since in this PR we are already adding detail explanation on how to deploy to custom network?

Reply:
I think I got confused about this. 

Adding a custom chain is adding a custom chain to the frontend right?

Reply:
> Adding a custom chain is adding a custom chain to the frontend right?

Yes yes, since thats what people struggle with just how to add chain to frontend. So I was saying we remove this section 3 completely checkout https://github.com/scaffold-eth/se-2-docs/pull/88#discussion_r1772556908